benchmarks/dynamo/common.py

```diff
@@ -247,30 +247,40 @@ def tensor_is_on_xla(tensors):
     return any(map(lambda x: x.device.type == "xla", tensors))
 
 
-def timed(model, model_iter_fn, example_inputs, times=1, return_result=False):
+def timed(model, model_iter_fn, example_inputs, times=1, return_result=False, collect_outputs=False):
+    use_xla = tensor_is_on_xla(example_inputs)
     synchronize()
-    if tensor_is_on_xla(example_inputs):
+
+    reset_rng_state()
+    if use_xla:
         import torch_xla.core.xla_model as xm
 
         xm.mark_step()
+        xm.wait_device_ops()
 
-    reset_rng_state()
     t0 = time.perf_counter()
     # Dont collect outputs to correctly measure timing
     for _ in range(times):
-        result = model_iter_fn(model, example_inputs, collect_outputs=False)
-        if tensor_is_on_xla(result):
+        if use_xla:
+            # force a const seed for xla
```

Comment:
why a const seed?  is it possible to move this logic to the main 'reset_rng_state` somehow to be consistent?  if not, i think its ok to do it here but i'd like to understand why you're doing it

Reply:
Non const seed is also ok as long as baseline and test are using the same seed. But I guess the 'reset_rng_state' also use some const seed like 1337, lol  I can change to use 1337 here as well, just let me know

Put it here inside the 'for' loop since we need reset the seed for each iteration.  We can not guarantee that the seed of baseline and test are sync after running the model because of extra `mark_step`s.

Reply:
i guess i was just thinking its better if we make a 'use_xla=False' arg to reset_rng_state fn.  then that fn can just do the reset for xla and `timed` doesn't have to add a second rng call

Reply:
Then I'll have to move `reset_rng_seed` into the loop due the reason I mentioned above. I guess that's fine since the reset seed calls should not be expensive and putting that inside the time measurement is not a big deal.